05_concurrency/homework/cache.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+type Cache interface {
+	Set(key string, value interface{}, ttl time.Duration)
+	Get(key string) (interface{}, bool)
+	Delete(key string)
+}
+
+type CacheInit struct {
+	mu sync.Mutex
+	v  map[string]Elem
+}
+
+type Elem struct {
+	val  interface{}
+	date time.Time
+	ttl  int64
+}
+
+func (c *CacheInit) Set(key string, value interface{}, ttl time.Duration) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.v[key] = Elem{
+		val:  value,
+		date: time.Now(),
+		ttl:  time.Now().Add(ttl).UnixNano(),
+	}
+}
+
+func (c *CacheInit) Get(key string) (interface{}, bool) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	val, ok := c.v[key]
+	return val.val, ok
+}
+
+func (c *CacheInit) Delete(key string) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	delete(c.v, key)
+}
+
+func (c *CacheInit) expiredKeys() (keys []string) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	for key, it := range c.v {
+		if time.Now().UnixNano() >= it.ttl {
+			keys = append(keys, key)
+		}
+	}
+	return
+}
+
+func creatingCache() *CacheInit {
```

Comment:
This function should be exported so others can import your package and use its types. Of course the name of package should not be `main` this way. Otherwise no one can initialize the map inside your cache implementation.

Reply:
Usually such function names start with `New`.
It can be 
```
package cache
func New()
```
or 

```
package store
func NewCache()
```
so it perfectly represents its usecase